Clean up HLS theta conversion

phate opened the pull request:
I have quite some problems with refactoring the HLS code. Start to clean it up such that I can familiarize myself better with the code and gain a better starting position for refactoring. This PR overhauls the theta conversion pass.

sjalander commented on jlm/hls/backend/rvsdg2rhls/theta-conv.cpp:
```diff
@@ -10,67 +10,72 @@
 
 void
-theta_conv(jlm::llvm::RvsdgModule & rm)
+theta_conv(jlm::llvm::RvsdgModule & rvsdgModule)
```
Why not update the function name?

sjalander replied:
Maybe call it ConvertAllTheta()

phate replied:
Renamed to ConvertThetaNodes.

sjalander commented on jlm/hls/backend/rvsdg2rhls/theta-conv.cpp:
```diff
@@ -10,67 +10,72 @@
+}
+
+static void
+ConvertThetaRegion(jlm::rvsdg::region & region)
```
This is the one that traverses the graphs and converts thetas, so it would make sense to call this one something with convert all thetas.

phate replied:
Renamed to ConvertThetaNodesInRegion

sjalander reviewed (commented):
I think the naming of the functions are missleading
ConvertThetaRegion is not taking a theta region and converting it but traverses the given region and finds all thetas that then get converted.
conv_theta() is not following the naming convention.
Previously, I've seen that you changed to explicitly name the function based on the argument type that is taken.

So maybe 
ConvertAllThetaInRegion()
ConvertAllThetaInModule()

phate commented on jlm/hls/backend/rvsdg2rhls/theta-conv.cpp:
```diff
@@ -10,67 +10,72 @@
+static void
+ConvertThetaStructuralNode(jlm::rvsdg::structural_node & structuralNode)
+{
+  for (size_t n = 0; n < structuralNode.nsubregions(); n++)
```
Renamed the functions to hopefully be clearer.

phate commented:
Changes since last review:
1. Renamed the functions
2. Renamed the files from theta-conv.hpp/*.cpp to ThetaConversion.hpp/*.cpp
3. Improved the documentation of ConvertThetaNodes function
